File: backend/src/main/java/com/gxs/bank/model/Promotion.java
package com.gxs.bank.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "promotions")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Promotion {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "badge_text")
    private String badgeText;

    @Column(name = "cta_text")
    private String ctaText;

    @Column(name = "cta_url")
    private String ctaUrl;

    @Column(name = "terms_url")
    private String termsUrl;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "valid_from")
    private LocalDate validFrom;

    @Column(name = "valid_to")
    private LocalDate validTo;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (isActive == null) isActive = true;
    }
}

File: backend-python/com/gxs/bank/model/Promotion.py
from datetime import date, datetime
import uuid

from sqlalchemy import Boolean, Column, Date, DateTime, String, Text

from com.gxs.bank.config.database import Base


class Promotion(Base):
    __tablename__ = "promotions"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    title = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    badgeText = Column("badge_text", String(255), nullable=True)
    ctaText = Column("cta_text", String(255), nullable=True)
    ctaUrl = Column("cta_url", String(1024), nullable=True)
    termsUrl = Column("terms_url", String(1024), nullable=True)
    imageUrl = Column("image_url", String(1024), nullable=True)
    validFrom = Column("valid_from", Date, nullable=True)
    validTo = Column("valid_to", Date, nullable=True)
    isActive = Column("is_active", Boolean, nullable=False, default=True)
    createdAt = Column("created_at", DateTime, nullable=False, default=datetime.utcnow)
